src/api.ts

```diff
@@ -7,16 +7,28 @@ const METRICS_ORIGIN = import.meta.env.DEV
   ? window.location.origin + "/metrics"
   : import.meta.env.VITE_METRICS_ORIGIN ?? "https://ln3tnkd4d5uiufjgimi6jlkmci0bceff.lambda-url.us-west-2.on.aws/";
 
+// The current setup for local testing uses a Lambda gateway that
+// does not parse the requests to lambda events.
+const IS_LOCAL_LAMBDA = import.meta.env.VITE_LOCAL_GATEWAY;
 /**
  * Fetch API wrapper that throws on 400+ http status.
  */
-export async function wfetch(resource: RequestInfo | URL, opts: RequestInit = {}) {
+export async function wfetch(resource: RequestInfo | URL, opts: RequestInit = {}, dataObj: object = {}) {
   if (Number.isFinite(opts.timeout)) {
     const controller = new AbortController();
     opts.signal = controller.signal;
     setTimeout(() => controller.abort(), opts.timeout);
   }
 
+  if (IS_LOCAL_LAMBDA) {
+    opts.headers = {
+      "Content-Type": "application/json",
+    };
+
+    opts.body = JSON.stringify(dataObj);
+    opts.method = "POST";
+  }
+
```

Comment:
this is pretty awkward

Reply:
I added this for running the lambda locally as a gateway and testing it. In that case, the response doesn't get parsed like AWS so had to add this awkward code. Would a better approach be to remove this code and add it in the README as local instructions for testing modifications to the lambda gateway?

Reply:
Ideally, this environment specific logic doesn't pollute the source code. 

We have 2 options

1. Modify the L1-dashboard dev environment 

The "best" approach would be to modify the proxy logic [here](https://github.com/filecoin-saturn/L1-dashboard/blob/b16631b9170162fc2ae419956d76b8d51d3503f0/vite.config.ts#L25-L32). 

docs: https://vitejs.dev/config/server-options.html#server-proxy

example: https://github.com/vitejs/vite/issues/6129

Implementation might look something like

```js
      proxy: {
        "/metrics": {
          target: env.VITE_METRICS_ORIGIN ?? "https://ln3tnkd4d5uiufjgimi6jlkmci0bceff.lambda-url.us-west-2.on.aws/",
          changeOrigin: true,
          secure: false,
          rewrite: (path) => path.replace(/^\/metrics/, ""),
          configure:  (proxy, options) => {
            proxy.on("proxyReq", function (proxyReq, req, res) {
                  console.log("proxy req for"+req.url);
                  // If req.url is to localhost (aka, not a lambda URL)
                  // Change method from GET to POST
                  // Move query parameters to request body.
            })
          },
        },
      },
```


To transparently change the method from `GET` to `POST`, and move any query parameters from the URL to the request body. This way the source code remains oblivious to the current environment, and there's no need to sprinkle special case logic.

2. Modify the log-ingestor dev environment.

How are you running the lambda locally? I don't see any code for it in the log-ingestor. You must be running an http server that listens for the POST request, right? And that server imports and calls the lambda handlers? You could add some proxy logic to the server that converts the query params into the lambda `event` object. This is likely easier than option 1.

Reply:
Thanks for the thorough explanation here. I think option 2 would be the most suitable to implement as well, and also replicated the actual prod environment the best.

In terms of how I run the lambda locally, I added some code in the `run-lambda.js` file which uses a library called `lambda-local` to spin up an http server as you mentioned.

I have removed the "awkward" code bits for now and opened an [issue](https://github.com/filecoin-saturn/log-ingestor/issues/60) in the log-ingestor repo to address the local lambda issues just to unblock this work for now if that is okay.

Reply:
ya sg